io/disk/htx_block_devices.py

```diff
@@ -0,0 +1,175 @@
+        """
+        Build 'HTX'.
+        """
+        self.distro_name = distro.detect().name
+        packages = ['git', 'gcc', 'make', 'libncurses5', 'g++', 'libdapl-dev',
+                    'ncurses-dev', 'libncurses-dev']
+        smm = SoftwareManager()
+        if self.distro_name == 'Ubuntu':
+            for pkg in packages:
+                if not smm.check_installed(pkg) and not smm.install(pkg):
+                    self.skip("Can not install %s" % pkg)
+        else:
+            self.skip("skipping for other than Ubuntu Distro")
+
+        htx_path = os.path.join(self.teststmpdir, "HTX/")
+        git.get_repo('https://github.com/open-power/HTX.git',
+                     destination_dir=htx_path)
+
+        build.run_make(htx_path, extra_args='all')
+        build.run_make(htx_path, extra_args='deb')
+        process.run('dpkg -r htxubuntu')
+        process.run('dpkg --purge htxubuntu')
+        process.run('dpkg -i htxubuntu.deb')
+        self.mdt_file = self.params.get('mdt_file', default='mdt.hd')
+        self.time_limit = self.params.get('time_limit', default=600)
+        self.block_device = self.params.get('disk', default=None)
+        if self.block_device is None:
+            self.skip("Needs the block devices to run the HTX")
+
+    def test(self):
+        """
+        Execute 'HTX' with appropriate parameters.
+        """
+        self.log.info("Starting the HTX Deamon")
+        process.run('/usr/lpp/htx/etc/scripts/htxd_run')
+
+        self.log.info("selecting the mdt file ")
+        cmd = "htxcmdline -select -mdt %s" % self.mdt_file
+        process.system(cmd, ignore_status=True)
+
+        if self.is_block_device_in_mdt() is False:
+            self.fail("Block devices %s are not available in %s",
+                      self.block_device, self.mdt_file)
+
+        self.suspend_all_block_device()
+
+        self.log.info("Activating the %s", self.block_device)
+        cmd = "htxcmdline -activate %s -mdt %s" % (self.block_device,
+                                                   self.mdt_file)
+        process.system(cmd, ignore_status=True)
+        if self.is_block_device_active() is False:
+            self.fail("Block devices failed to activate")
+
+        self.log.info("Running the HTX on %s", self.block_device)
+        cmd = "htxcmdline -run %s -mdt %s" % (self.block_device, self.mdt_file)
+        process.system(cmd, ignore_status=True)
+        for _ in range(0, self.time_limit, 60):
+            self.log.info("Error logs")
+            process.run('htxcmdline -geterrlog')
+            if os.stat('/tmp/htxerr').st_size != 0:
+                self.fail("check errorlogs for exact error and failure")
+            self.log.info("status of block devices after every 60 sec")
+            cmd = 'htxcmdline -query %s -mdt %s' % (self.block_device,
+                                                    self.mdt_file)
+            process.system(cmd, ignore_status=True)
+            time.sleep(60)
```

Comment:
@PraveenPenguin This sleep is required so as to check every minute. Else, we would be checking every second